test/integration/red_test_client.go

```diff
@@ -0,0 +1,53 @@
+//go:build integration
+
+package integration
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/mariomac/guara/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/ebpf-autoinstrument/test/integration/components/prom"
+)
+
+func testREDMetricsForClientHTTPLibrary(t *testing.T) {
+	// Eventually, Prometheus would make this query visible
+	pq := prom.Client{HostPort: prometheusHostPort}
+	var results []prom.Result
+	test.Eventually(t, testTimeout, func(t require.TestingT) {
+		var err error
+		results, err = pq.Query(`http_client_duration_seconds_count{` +
+			`http_method="GET",` +
+			`http_status_code="200",` +
+			`service_namespace="integration-test",` +
+			`service_name="pingclient"}`)
+		require.NoError(t, err)
+		// check duration_count has 3 calls and all the arguments
```

Comment:
Seems like the details in the comment are stale from copy/paste from other code.